subsys/net/lib/openthread/platform/radio.c

```diff
@@ -272,6 +280,14 @@ void transmit_message(struct k_work *tx_job)
 	radio_api->set_channel(radio_dev, sTransmitFrame.mChannel);
 	radio_api->set_txpower(radio_dev, tx_power);
 
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
+	tx_pkt->ieee802154_time_ie_offset = sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset;
```

Comment:
What about implementing `net_pkt_set_ieee802154_time_ie_offset()` and `net_pkt_set_ieee802154_network_time_offset()`?